ispyb/connector/mysqlsp/main.py

```diff
@@ -61,17 +53,13 @@ def get_data_area_package(self):
     return 'ispyb.sp'
 
   def create_cursor(self, dictionary=False):
-      if time.time() - self.last_activity_ts > self.conn_inactivity:
-          # re-connect:
-          self.connect(self.user, self.pw, self.host, self.db, self.port)
-      self.last_activity_ts = time.time()
-      if self.conn is None:
-          raise ISPyBConnectionException
-
-      cursor = self.conn.cursor(dictionary=dictionary)
-      if cursor is None:
-          raise ISPyBConnectionException
-      return cursor
+    if not self.conn:
+      raise ISPyBConnectionException('Not connected to database')
+    self.conn.ping(reconnect=True)
```

Comment:
Maybe it would be useful to try and catch the `InterfaceError` and do something with it? At least throw our own ISPyBConnectionException? Would it make sense to retry the connect if the initial ping reconnect doesn't work? Maybe use parameters like `ping(reconnect=True, attempts=6, delay=1)` ? If so, should probably do the same for the initial `connect` as well.

Reply:
```ping()``` only raises an ```InterfaceError``` if ```reconnect=False```, so this should not be a problem.
I agree on the other point. I think attempts and delay could be set globally in the configuration file for both connect and reconnect. I would do that in a separate issue though.